yt_dlp/networking/exceptions.py

```diff
@@ -75,7 +75,7 @@ def __repr__(self):
 
 
 class IncompleteRead(TransportError):
-    def __init__(self, partial: int, expected: int = None, **kwargs):
+    def __init__(self, partial: int, expected: int | None = None, **kwargs):
```

Comment:
Isn't this already implied by `the = None`?

Reply:
Not in my type checker. I got an error for this

Reply:
Right, might've got mixed up. I think you need `from __future__ import annotations` too since this annotation is py3.10+
https://docs.python.org/3/library/typing.html#typing.Optional